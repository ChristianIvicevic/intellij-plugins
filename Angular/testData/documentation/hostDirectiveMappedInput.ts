import {Component, Directive, Input} from '@angular/core';

// @ts-ignore
@Directive({
   selector: '[appBold]',
   standalone: true,
   exportAs: "bold"
 })
export class BoldDirective {
  @Input()
  weight: Number = 12
}

@Component({
   standalone: true,
   selector: 'app-test',
   template: `
      <app-test font<caret>Weight='13'></app-test>
   `,
   hostDirectives: [{
     directive: BoldDirective,
     inputs: ['weight: fontWeight']
   }],
 })
export class TestComponent  {
}
