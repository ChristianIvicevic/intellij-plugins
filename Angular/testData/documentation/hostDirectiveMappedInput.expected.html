<div class='definition'><pre><span style='color: #000080'>directive input</span> <span style='color: #660e7a'>weight</span><span style='color: #000000'>: </span><span style='color: #000000'><span style=""><span style='color: #589df6'><a href="psi_element://Number">Number</a></span></span></span></pre></div>
<div class='definition'><pre><span style='color: #000080'>standalone</span> <span style='color: #000080'>directive</span> <span style='color: #000000'><span style="">BoldDirective</span></span></pre></div><table class='sections'><tr><td valign='top' class='section'><p>Selectors:</td><td valign='top'><p><pre><span style="">[</span><span style="color:#000080;">appBold</span><span style="">]</span></pre></td><tr><td valign='top' class='section'><p>Export as:</td><td valign='top'><p><pre><span style='color: #660e7a'>bold</span></pre></td></table>